static/js/src/app/webhooks.js

```diff
@@ -0,0 +1,182 @@
+                        </button>
+                        <button class="btn btn-primary edit_button" data-toggle="modal" data-backdrop="static" data-target="#modal" data-webhook-id="${webhook.id}">
+                          <i class="fa fa-pencil"></i>
+                        </button>
+                        <button class="btn btn-danger delete_button" data-webhook-id="${webhook.id}">
+                          <i class="fa fa-trash-o"></i>
+                        </button>
+                      </div>
+                    `
+                ]).draw()
+            })
+        })
+        .error(() => {
+            errorFlash("Error fetching webhooks")
+        })
+};
+
+const editWebhook = (id) => {
+    $("#modalSubmit").unbind("click").click(() => {
+        saveWebhook(id);
+    });
+    if (id !== -1) {
+        api.webhookId.get(id)
+          .success(function(wh) {
+              $("#name").val(wh.name);
+              $("#url").val(wh.url);
+              $("#secret").val(wh.secret);
+              $("#is_active").prop("checked", wh.is_active);
+          })
+          .error(function () {
+              errorFlash("Error fetching webhook")
+          });
+    }
+};
+
+const deleteWebhook = (id) => {
+    var wh = webhooks.find(x => x.id == id);
+    if (!wh) {
+        return;
+    }
+    Swal.fire({
+        title: "Are you sure?",
+        text: `This will delete the webhook '${wh.name}'`,
+        type: "warning",
+        animation: false,
+        showCancelButton: true,
+        confirmButtonText: "Delete",
+        confirmButtonColor: "#428bca",
+        reverseButtons: true,
+        allowOutsideClick: false,
+        preConfirm: function () {
+            return new Promise((resolve, reject) => {
+                api.webhookId.delete(id)
+                    .success((msg) => {
+                        resolve()
+                    })
+                    .error((data) => {
+                        reject(data.responseJSON.message)
+                    })
+            })
+            .catch(error => {
+                Swal.showValidationMessage(error)
+              })
+        }
+    }).then(function(result) {
+        if (result.value) {
+            Swal.fire(
+                "Webhook Deleted!",
+                `The webhook has been deleted!`,
+                "success"
+            );
+        }
+        $("button:contains('OK')").on("click", function() {
+            location.reload();
+        })
+    })
+};
+
+const pingUrl = (btn, whId) => {
+    dismiss();
+    btn.disabled = true;
+    api.webhookId.ping(whId)
+        .success(function(wh) {
+            btn.disabled = false;
+            successFlash(`Ping of "${wh.name}" webhook succeeded.`);
+        })
+        .error(function() {
+            btn.disabled = false;
+            var wh = webhooks.find(x => x.id == whId);
+            if (!wh) {
+                return
+            }
+            errorFlash(`Ping of "${wh.name}" webhook failed.`)
```

Comment:
If we're returning an error from the webhook ping, we should display it to the user here.

Reply:
fixed